pkg/step/engine.go

```diff
@@ -88,3 +88,16 @@ func (c *Engine) AddToQueue(workflow string, record interface{}) (err error) {
 	err = addErr
 	return
 }
+
+func (c *Engine) SearchForRecords(filter []byte, cacheName string) ([]*RawRecord, error) {
```

Comment:
This method is supposed to be more user friendly. This is where they should pass in a`filter interface{}` similar to how the mongo driver `Find` method works. You can pass bson, bytes or a struct and it will do the right thing.
You can see how the Mongo driver handles this here:

https://github.com/mongodb/mongo-go-driver/blob/master/mongo/collection.go#L938

I also think we should model the naming after Mongo as well.

`Find` and `FindOne`

Reply:
updated